activation/listeners.go

```diff
@@ -37,6 +37,24 @@ func Listeners(unsetEnv bool) ([]net.Listener, error) {
 	return listeners, nil
 }
 
+// ListenersWithNames maps a listener name to a set of net.Listener instances.
+func ListenersWithNames(unsetEnv bool) (map[string][]net.Listener, error) {
+	files := Files(unsetEnv)
+	listeners := map[string][]net.Listener{}
+
+	for _, f := range files {
+		if pc, err := net.FileListener(f); err == nil {
```

Comment:
I think this would benefit from the same conditional `Close` we are discussing at https://github.com/coreos/go-systemd/pull/250.